bravado/client.py

```diff
@@ -102,11 +102,23 @@ def from_url(cls, spec_url, http_client=None, request_headers=None,
         :param config: Config dict for bravado and bravado_core.
             See CONFIG_DEFAULTS in :module:`bravado_core.spec`.
             See CONFIG_DEFAULTS in :module:`bravado.client`.
+
+        :rtype: :class:`bravado_core.spec.Spec`
         """
         log.debug(u"Loading from %s" % spec_url)
         http_client = http_client or RequestsClient()
         loader = Loader(http_client, request_headers=request_headers)
         spec_dict = loader.load_spec(spec_url)
+
+        # RefResolver may have to download additional json files (remote refs)
+        # via http. Wrap http_client's request() so that request headers are
+        # passed along with the request transparently. Yeah, this is not ideal,
+        # but since RefResolver has new found responsibilities, it is
+        # functional.
```

Comment:
I have few doubts:
1) So, `spec_dict` at this point may have external refs present, correct?
2) What are the points when additional json files would need to be downloaded  (i suppose it is done in bravado-core and this http_client is fed to it)?
3) Instead of adding this wrapper, is it possible to use a clone of this `http_client` and use that instead in bravado-core?